### ISSUE TITLE
Terra Date Picker isn't focusable or actionable with a screen reader.

### ISSUE THREAD

**Issue by JakeLaCombe** [cerner/terra-core#1828]
# Feature Request

## Description
Terra Datepicker is hard to use with a screen reader.

1. Focusing and clicking the button doesn't auto focus to the datepicker.
2. The calendar doesn't have descriptive text when used with a screen reader.

# Additional Details
It is discussed that the accessible workflow would be just having them use the text input. Perhaps this could be a solution?

**Comment by JakeLaCombe** [comment]
We talked in leadership today about our solution for this.

1. Screen Reader pattern would be to use the input instead of the visual datepicker.
2. Redo the input to have a pattern similar to time input.

**Comment by bjankord** [comment]
> Focusing and clicking the button doesn't auto focus to the datepicker.
The calendar doesn't have descriptive text when used with a screen reader.

These are also noticable when using VoiceOver on iOS. I am able to complete all the tasks from manual date entry in the input to selecting a year, month, and day in the date picker with VO, but there is a lack of descriptive text when using the datepicker popup. Have descript text to label the year and month selects would be helpful as well as descriptive text that could communicate the day abbreviations. 

It would also be helpful to have the full date read to the user as they interact with the days in the calendar. This would be a much larger task as we'd be looking into a fair amount of translation work to acheive this functionality.

**Comment by bjankord** [comment]
Excluded/filtered dates do not indicate that they are excluded or filtered (disabled) to screen reader users.

**Comment by bjankord** [comment]
It may also be nice to add a clear way to close the datepicker popup with out selecting a date when using VoiceOver on iOS. I was able to do this by shifting focus back to the input and selecting it but having a clear close datepicker button would be helpful for users interacting with the datepicker with this setup.
